src/log.rs

```diff
@@ -68,10 +70,41 @@ pub trait LogExt<T> {
     /// - You won't get any warnings about unused results but can still use the value if you need it
     /// - This prevents the same warning from being printed to the log multiple times
     ///
+    /// Example: This:
+    /// ```
+    /// if let Err(e) = do_something() {
+    ///     warn!(context, "{:#}", e);
+    /// }
+    /// ```
+    /// can be replaced with:
+    /// ```
+    /// do_something().log(context);
+    /// ```
+    ///
     /// Thanks to the [track_caller](https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html#track_caller)
     /// feature, the location of the caller is printed to the log, just like with the warn!() macro.
+    ///
+    /// Unfortunately, the track_caller feature does not work on async functions (as of Rust 1.50).
+    /// This means that we can't make our logs even better by adding `#[track_caller]` to our helper
+    /// functions.  
+    /// See https://github.com/rust-lang/rust/issues/78840 for progress on this.
     #[track_caller]
     fn log(self, context: &Context) -> Option<T>;
+
+    /// Like `log()`, but you can pass an extra string message.
+    ///
+    /// Example: This:
+    /// ```
+    /// if let Err(e) = do_something() {
+    ///     warn!(context, "Something went wrong: {:#}", e);
+    /// }
+    /// ```
+    /// can be replaced with:
+    /// ```
+    /// do_something().log_m(context, "Something went wrong");
+    /// ```
+    #[track_caller]
+    fn log_m(self, context: &Context, msg: &str) -> Option<T>;
```

Comment:
Other possible names would be:
- `log_c`, c for "context" (as you are giving additional context, just as with the anyhow `.context()` method)
- `log_s`, s for "str" (well, there is an extra string argument)

Reply:
Give that we use anyhow pretty much anywhere, would we be able to do `.context("oops").log_err()` instead of needing a new API?